### DISCUSSION TITLE
First release

### DISCUSSION THREAD

**Discussion by merlinschumacher** [merlinschumacher/Open-Covid-Certificate-Validator#4] [Announcements]
The first release contains a working validator for EU COVID certificates / EU Green Certificates / Digitaler Impfnachweis. It currently only supports the certificate list provided by Germany, which should be able to validate all conforming European vaccination certificates.
The German certificates are signature checked when (down)loaded. 

_Notice: This software is still very rudimentary. It comes withour any warrant._

<hr /><em>This discussion was created from the release <a href='https://github.com/merlinschumacher/Open-Covid-Certificate-Validator/releases/tag/v0.0.1'>First release</a>.</em>

**Comment by ffischer1** [comment]
Very nice project, but there is one thing I don't understand yet: is only the structure and the certificate chain for a covid certificate (qr code) checked or is the test date also taken into account? For example, will a 3 days old test be rejected as not valid? Or a vaccination certificate that is still missing a vaccination, will be also shown as invalid?

**Reply by merlinschumacher** [reply]
Currently, only the certificate is decoded and this it's cryptographic key validated, but the contents of the cert are not evaluated. A 3 day old certificate would be valid to the service, so would be one that is only the first in a row of two injections. This would be an improvement for the next version, but might require some effort, since I'm not how consistent the rules are allover the EU member states.
